client/webserver/site/src/js/markets.ts

```diff
@@ -1837,6 +1853,19 @@ export default class MarketsPage extends BasePage {
     this.setRegistrationStatusVisibility()
   }
 
+  /*
+   * handleMatchNote is the handler for the 'match'-type notification, which are
+   * used to update a user's match's status.
+   */
+  handleMatchNote (note: MatchNote) {
+    const market = this.market
+    const orders = app().orders(market.dex.host, marketID(market.baseCfg.symbol, market.quoteCfg.symbol))
+    const order = orders.find((ord: Order) => ord.id === note.orderID)
+    if (!order) return
```

Comment:
The only thing that would change on this page due to a match note is showing/hiding the acceleration button. The only change you need is adding this new `handleMatchNote` function, and adding the code below:
``` js
    const mord = this.metaOrders[order.id]
    if (!mord) return this.refreshActiveOrders()
    mord.ord = order
    if (app().canAccelerateOrder(order)) Doc.show(mord.details.accelerateBttn)
    else Doc.hide(mord.details.accelerateBttn)
```

The better way to solve this would probably be to send an order note when the order becomes able to be accelerated, or stops being able to be accelerated. I think that actually makes more sense, since we're dealing with orders on this page, and this is a situation where the state of an order has changed.

Reply:
Thanks for the suggestion @martonp . I am now sending a new order notification after mining a block.

Right now I am keeping the match notification, because it is useful on the order page. 

Do you think would it be good to remove the match notification and keep only the new order notification when confirming new blocks?

Reply:
If we go with the backend solution, consider:

An order **becomes** able to be accelerated when user's own swap/init is just broadcasted. For both maker and taker, it looks like an order note is already sent after the call to `c.swapMatches(t, swaps)` in `tick`.  :heavy_check_mark: 

An order **stops** being able to be accelerated when the user's own swap/init first reaches 1 confirm.
- For maker, this is detected in `isSwappable` in the `case order.MakerSwapCast:` with `match.Side == order.Maker`.
- For taker, this is detected in `isRedeemable`in the `case order.TakerSwapCast` with `match.Side == order.Taker`.

The counterparty's swap or redeem is not relevant in either case.

So it seems like your previous commit was more correct.  I think what is missing is just a condition to only send the order note when it hits 1 conf.

Does it appear to be working correctly with your latest revision.  If so, I'm confused, because confs on the counterparty's swap does not seem relevant to this.

Reply:
> I think what is missing is just a condition to only send the order note when it hits 1 conf.

Regarding this, perhaps a small modification to `(*matchTracker).setSwapConfirms`:

```diff
diff --git a/client/core/trade.go b/client/core/trade.go
index 0cca3002..ec7da7ab 100644
--- a/client/core/trade.go
+++ b/client/core/trade.go
@@ -164,8 +164,8 @@ func (m *matchTracker) confirms() (mine, theirs int64) {
        return atomic.LoadInt64(&m.swapConfirms), atomic.LoadInt64(&m.counterConfirms)
 }
 
-func (m *matchTracker) setSwapConfirms(mine int64) {
-       atomic.StoreInt64(&m.swapConfirms, mine)
+func (m *matchTracker) setSwapConfirms(mine int64) (was int64) {
+       return atomic.SwapInt64(&m.swapConfirms, mine)
 }
 
 func (m *matchTracker) setCounterConfirms(theirs int64) (was int64) {
@@ -1283,10 +1283,12 @@ func (t *trackedTrade) isSwappable(ctx context.Context, match *matchTracker) (re
                                "but we have not seen the counter-party's redemption yet. This could just"+
                                " be network latency.", match)
                }
-               match.setSwapConfirms(int64(confs))
-               cord := t.coreOrder()
-               t.notify(newOrderNote(TopicConfirms, "", "", db.Data, cord))
-               t.notify(newMatchNote(TopicConfirms, "", "", db.Data, t, match))
+               was := match.setSwapConfirms(int64(confs))
+               if was == 0 && confs > 0 {
+                       t.notify(newOrderNote(TopicConfirms, "", "", db.Data, t.coreOrderInternal())) // includes all current match data
+               } else {
+                       t.notify(newMatchNote(TopicConfirms, "", "", db.Data, t, match))
+               }
                return false, false
        }
 
@@ -1514,10 +1516,12 @@ func (t *trackedTrade) isRedeemable(ctx context.Context, match *matchTracker) (r
                                "but we have not seen the counter-party's redemption yet. This could just"+
                                " be network latency.", match)
                }
-               match.setSwapConfirms(int64(confs))
-               cord := t.coreOrder()
-               t.notify(newOrderNote(TopicConfirms, "", "", db.Data, cord))
-               t.notify(newMatchNote(TopicConfirms, "", "", db.Data, t, match))
+               was := match.setSwapConfirms(int64(confs))
+               if was == 0 && confs > 0 {
+                       t.notify(newOrderNote(TopicConfirms, "", "", db.Data, t.coreOrderInternal())) // includes all current match data
+               } else {
+                       t.notify(newMatchNote(TopicConfirms, "", "", db.Data, t, match))
+               }
                return false, false
 
        case order.MakerRedeemed:
```

Not tested, just thinking.

Reply:
Still, an order note that includes all match data feels a little brute force.  I'm not totally convinced that this is better than updating the frontend's match note handling as per https://github.com/decred/dcrdex/pull/1847#discussion_r1001917323

Reply:
I believe what @martonp was getting at is this:

```diff
diff --git a/client/webserver/site/src/js/markets.ts b/client/webserver/site/src/js/markets.ts
index e0eebf21..0a520319 100644
--- a/client/webserver/site/src/js/markets.ts
+++ b/client/webserver/site/src/js/markets.ts
@@ -50,6 +50,7 @@ import {
   SpotPriceNote,
   FeePaymentNote,
   OrderNote,
+  MatchNote,
   EpochNote,
   BalanceNote,
   MiniOrder,
@@ -471,6 +472,7 @@ export default class MarketsPage extends BasePage {
     // Notification filters.
     app().registerNoteFeeder({
       order: (note: OrderNote) => { this.handleOrderNote(note) },
+      match: (note: MatchNote) => { this.handleMatchNote(note) },
       epoch: (note: EpochNote) => { this.handleEpochNote(note) },
       conn: (note: ConnEventNote) => { this.handleConnNote(note) },
       balance: (note: BalanceNote) => { this.handleBalanceNote(note) },
@@ -1860,6 +1862,17 @@ export default class MarketsPage extends BasePage {
       (oldStatus === OrderUtil.StatusBooked && order.status > OrderUtil.StatusBooked)) this.setDepthMarkers()
   }
 
+  /* handleMatchNote handles a 'match' notification. The sole purpose of this is
+   * to show or hide the accelerate order icon, which depends on the status of
+   * an order's matches.
+   */
+  handleMatchNote (note: MatchNote) {
+    const mord = this.metaOrders[note.orderID]
+    if (!mord) return this.refreshActiveOrders()
+    if (app().canAccelerateOrder(mord.ord)) Doc.show(mord.details.accelerateBttn)
+    else Doc.hide(mord.details.accelerateBttn)
+  }
+
   /*
    * handleEpochNote handles notifications signalling the start of a new epoch.
    */
```

Reply:
I don't think we should send full order notes for this anymore though.  I think we should just handle the match notes that are already sent to the UI properly on the /markets, as we do on the /order page.  https://github.com/decred/dcrdex/pull/1847#discussion_r1007414247